src/segments/argocd_test.go

```diff
@@ -0,0 +1,233 @@
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		if len(tc.ExpectedError) > 0 {
+			_, err := argocd.getConfigFromOpts()
+			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
+		} else {
+			config, err := argocd.getConfigFromOpts()
+			assert.NoErrorf(t, err, tc.Case)
+			assert.Equal(t, tc.Expected, config, tc.Case)
+		}
+	}
+}
+
+func TestArgocdGetConfigPath(t *testing.T) {
+	home := "/Users/posh"
+	configFile := path.Join(home, ".config", "argocd", "config")
+	cases := []struct {
+		Case          string
+		Opts          string
+		Expected      string
+		ExpectedError string
+	}{
+		{Case: "without opts", Expected: configFile},
+		{Case: "with opts", Opts: "--config /etc/argocd/config", Expected: "/etc/argocd/config"},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("Home").Return(home)
+		env.On("Getenv", argocdOptsEnv).Return(tc.Opts)
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		assert.Equal(t, tc.Expected, argocd.getConfigPath())
+	}
+}
+
+func TestArgocdParseConfig(t *testing.T) {
+	configFile := "../test/argocd.yml"
+	cases := []struct {
+		Case            string
+		Config          string
+		Expected        bool
+		ExpectedError   string
+		ExpectedContext ArgocdContext
+	}{
+		{Case: "missing or empty yaml", Config: "", ExpectedError: argocdInvalidYaml},
+		{Case: "invalid yaml", Config: `
+[context]
+context
+`, ExpectedError: argocdInvalidYaml},
+		{Case: "invalid config", Config: `
+contexts:
+  - name: context1
+    server: server1
+    user: user1
+  - name: context2
+    server: server2
+    userr: user2
+current-context: context2
+servers:
+  - grpc-web: true
+    server: server1
+  - grpc-web: false
+    server: serve2
+`, ExpectedError: argocdInvalidYaml},
+		{Case: "no current context found", Config: `
+contexts:
+  - name: context1
+    server: server1
+    user: user1
+  - name: context2
+    server: server2
+    user: user2
+`, ExpectedError: argocdNoCurrent},
+		{Case: "current context found", Expected: true, Config: `
+contexts:
+  - name: context1
+    server: server1
+    user: user1
+  - name: context2
+    server: server2
+    user: user2
+current-context: context2
+servers:
+  - grpc-web: true
+    server: server1
+  - grpc-web: false
+    server: serve2
+users:
+  - auth-token: authtoken1
+    name: user1
+    refresh-token: refreshtoken1
+  - auth-token: authtoken2
+    name: user2
+    refresh-token: refreshtoken2
+`, ExpectedContext: ArgocdContext{
+			Name:   "context2",
+			Server: "server2",
+			User:   "user2",
+		}},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("FileContent", configFile).Return(tc.Config)
+		env.On("Error", mock2.Anything).Return()
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		if len(tc.ExpectedError) > 0 {
+			_, err := argocd.parseConfig(configFile)
+			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
```

Comment:
early return instead of if/else

```suggestion
			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
			continue
```

Reply:
updated with early return.